rest/data/project.go

```diff
@@ -68,6 +78,24 @@ func (pc *DBProjectConnector) UpdateProject(projectRef *model.ProjectRef) error
 	return nil
 }
 
+func (sc *DBProjectConnector) VerifyUniqueProject(name string) error {
+	_, err := sc.FindProjectById(name, false)
+	if err == nil {
```

Comment:
It seems to me like it would be possible to get an error that's not a statusNotfound error. In that case we would assume that we are good to go ahead and create the project (since err != nil), but that error may be some other error, not an error due to the project not being found.